pkg/steps/lease.go

```diff
@@ -136,3 +139,12 @@ func heartbeatNamespace(namespace func() string, client coreclientset.NamespaceI
 		}
 	}
 }
+
+func printResourceMetrics(client lease.Client, rtype string) {
+	m, err := client.Metrics(rtype)
+	if err != nil {
+		log.Printf("warning: Could not get resource metrics: %v", err)
+		return
+	}
+	log.Printf("warning: Failed to acquire resource, current capacity: %d free, %d leased", m.Free, m.Leased)
```

Comment:
IMO this is not a "warning" anymore, just "Failed to ..." makes sense on its own. Also, can we say "Failed to acquire resource in Xm time window." or something similar to indicate how long we attempted to get a lease (to avoid implying we checked just once).

I'm a little worried that we'll get pinged about races where we first give up, then something will return a lease, then we get the metric, say something like `1 free, 149 lease`, but other than that I generally like showing the metrics.

Reply:
Changed to `error: …` to follow the format of other `log` messages.

Reply:
> Also, can we say "Failed to acquire resource in Xm time window." or something similar to indicate how long we attempted to get a lease (to avoid implying we checked just once).

That is usually deducible from the time stamps.  In the original sample output, I used a 5s timeout for testing:

```
2020/09/04 18:08:42 Acquiring lease for "aws-quota-slice"
2020/09/04 18:08:47 warning: Failed to acquire resource, current capacity: 0 free, 1 leased
```

Do you think that is enough?

> I'm a little worried that we'll get pinged about races where we first give up, then something will return a lease, then we get the metric, say something like `1 free, 149 lease`, but other than that I generally like showing the metrics.

I considered that, but it should be exceedingly rare (although, with a code base such as `boksos`', you never know).  I think we can adjust later if it manifests frequently.